comp/forwarder/defaultforwarder/params.go

```diff
@@ -6,36 +6,38 @@
 package defaultforwarder
 
 import (
-	"github.com/DataDog/datadog-agent/comp/core/config"
-	log "github.com/DataDog/datadog-agent/comp/core/log/def"
-	"github.com/DataDog/datadog-agent/comp/forwarder/defaultforwarder/resolver"
-	"github.com/DataDog/datadog-agent/pkg/config/utils"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
 )
 
 // Params contains the parameters to create a forwarder.
 type Params struct {
-	UseNoopForwarder bool
-	// TODO: (components) When the code of the forwarder will be
-	// in /comp/forwarder move the content of forwarder.Options inside this struct.
-	Options *Options
+	useNoopForwarder              bool
+	withResolver                  bool
+	disableAPIKeyCheckingOverride optional.Option[bool]
+	features                      []Features
 }
 
 // NewParams initializes a new Params struct
-func NewParams(config config.Component, log log.Component) Params {
-	return Params{Options: NewOptions(config, log, getMultipleEndpoints(config, log))}
+func NewParams() Params {
+	return Params{withResolver: false}
 }
 
 // NewParamsWithResolvers initializes a new Params struct with resolvers
-func NewParamsWithResolvers(config config.Component, log log.Component) Params {
-	keysPerDomain := getMultipleEndpoints(config, log)
-	return Params{Options: NewOptionsWithResolvers(config, log, resolver.NewSingleDomainResolvers(keysPerDomain))}
+func NewParamsWithResolvers() Params {
```

Comment:
Should we return a pointer to Params? Since we are later using the pointer semantic functions, I think it makes sense to return a pointer 😄 

Same for `NewParams`

Reply:
In terms or ergonomic, why did you decided to create a function `NewParamsWithResolvers` instead of 
`func (p *Params) UseResolvers()`?

Reply:
Other `Params` functions are returned by copied.
I didn't decide to create the function `NewParamsWithResolvers`(it already exists).

Reply:
For now I kept the code as it is.